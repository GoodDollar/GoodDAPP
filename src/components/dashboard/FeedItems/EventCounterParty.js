import React from 'react'
import { Text } from '../../common'
import { withStyles } from '../../../lib/styles'

const EventCounterParty = ({ feedItem, styles, style, subtitle }) => {
  const direction =
    feedItem.type === 'send'
      ? 'To:'
      : ['claim', 'receive', 'withdraw', 'bonus'].indexOf(feedItem.type) > -1
      ? 'From:'
      : ''
  return (
    <Text textTransform="capitalize" textAlign="left" style={style} numberOfLines={1} ellipsizeMode="tail">
      {direction && (
        <Text fontSize={10} style={styles.direction}>
          {direction}
        </Text>
      )}
      <Text fontWeight="medium" lineHeight={19} style={styles.fullName}>
        {feedItem.data.subtitle && subtitle ? ` ${feedItem.data.subtitle}` : ` ${feedItem.data.endpoint.fullName}`}
      </Text>
    </Text>
  )
}

const getStylesFromProps = ({ theme }) => ({
  direction: {
    textAlignVertical: 'middle',
    marginRight: 3,
  },
  fullName: {
    textAlignVertical: 'middle',
  },
})

export default withStyles(getStylesFromProps)(EventCounterParty)
