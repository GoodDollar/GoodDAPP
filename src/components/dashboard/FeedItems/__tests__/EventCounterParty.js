import React from 'react'
import renderer from 'react-test-renderer'
import EventCounterParty from '../EventCounterParty'
import { mockEvent } from '../../__tests__/__util__'
import { withThemeProvider } from '../../../../__tests__/__util__'

// Note: test renderer must be required after react-native.

describe('EventCounterParty', () => {
  const WrappedEventCounterParty = withThemeProvider(EventCounterParty)

  describe('withdraw', () => {
    const feedItem = mockEvent('withdraw')

    it('matches snapshot', async () => {
      let tree

      // eslint-disable-next-line require-await
      await renderer.act(async () => (tree = renderer.create(<WrappedEventCounterParty feedItem={feedItem} />)))
      expect(tree.toJSON()).toMatchSnapshot()
    })
  })

  describe('send', () => {
    const feedItem = mockEvent('send')

    it('matches snapshot', async () => {
      let tree

      // eslint-disable-next-line require-await
      await renderer.act(async () => (tree = renderer.create(<WrappedEventCounterParty feedItem={feedItem} />)))
      expect(tree.toJSON()).toMatchSnapshot()
    })
  })

  describe('receive', () => {
    const feedItem = mockEvent('receive')

    it('matches snapshot', async () => {
      let tree

      // eslint-disable-next-line require-await
      await renderer.act(async () => (tree = renderer.create(<WrappedEventCounterParty feedItem={feedItem} />)))
      expect(tree.toJSON()).toMatchSnapshot()
    })
  })
})
