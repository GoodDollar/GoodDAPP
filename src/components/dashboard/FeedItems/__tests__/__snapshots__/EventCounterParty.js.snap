// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventCounterParty receive matches snapshot 1`] = `
<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "WebkitBoxFlex": 1,
      "WebkitFlexBasis": "0%",
      "WebkitFlexGrow": 1,
      "WebkitFlexShrink": 1,
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
      "msFlexNegative": 1,
      "msFlexPositive": 1,
      "msFlexPreferredSize": "0%",
    }
  }
>
  <div
    className="css-text-901oao css-textOneLine-bfa6kz"
    dir="auto"
    style={
      Object {
        "color": "rgba(66,69,74,1.00)",
        "direction": "ltr",
        "fontFamily": "Roboto",
        "fontSize": "16px",
        "fontWeight": "normal",
        "lineHeight": "22px",
        "textAlign": "left",
        "textDecoration": "none",
        "textTransform": "capitalize",
      }
    }
  >
    <span
      className="css-text-901oao css-textHasAncestor-16my406"
      dir="auto"
      style={
        Object {
          "color": "rgba(66,69,74,1.00)",
          "direction": "ltr",
          "fontFamily": "Roboto",
          "fontSize": "10px",
          "fontWeight": "normal",
          "lineHeight": "14px",
          "marginRight": "3px",
          "textAlign": "center",
          "textDecoration": "none",
          "textTransform": "none",
          "verticalAlign": "middle",
        }
      }
    >
      From:
    </span>
    <span
      className="css-text-901oao css-textHasAncestor-16my406"
      dir="auto"
      style={
        Object {
          "color": "rgba(66,69,74,1.00)",
          "direction": "ltr",
          "fontFamily": "Roboto",
          "fontSize": "16px",
          "fontWeight": "500",
          "lineHeight": "19px",
          "textAlign": "center",
          "textDecoration": "none",
          "textTransform": "none",
          "verticalAlign": "middle",
        }
      }
    >
       Misao Matimbo
    </span>
  </div>
</div>
`;

exports[`EventCounterParty send matches snapshot 1`] = `
<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "WebkitBoxFlex": 1,
      "WebkitFlexBasis": "0%",
      "WebkitFlexGrow": 1,
      "WebkitFlexShrink": 1,
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
      "msFlexNegative": 1,
      "msFlexPositive": 1,
      "msFlexPreferredSize": "0%",
    }
  }
>
  <div
    className="css-text-901oao css-textOneLine-bfa6kz"
    dir="auto"
    style={
      Object {
        "color": "rgba(66,69,74,1.00)",
        "direction": "ltr",
        "fontFamily": "Roboto",
        "fontSize": "16px",
        "fontWeight": "normal",
        "lineHeight": "22px",
        "textAlign": "left",
        "textDecoration": "none",
        "textTransform": "capitalize",
      }
    }
  >
    <span
      className="css-text-901oao css-textHasAncestor-16my406"
      dir="auto"
      style={
        Object {
          "color": "rgba(66,69,74,1.00)",
          "direction": "ltr",
          "fontFamily": "Roboto",
          "fontSize": "10px",
          "fontWeight": "normal",
          "lineHeight": "14px",
          "marginRight": "3px",
          "textAlign": "center",
          "textDecoration": "none",
          "textTransform": "none",
          "verticalAlign": "middle",
        }
      }
    >
      To:
    </span>
    <span
      className="css-text-901oao css-textHasAncestor-16my406"
      dir="auto"
      style={
        Object {
          "color": "rgba(66,69,74,1.00)",
          "direction": "ltr",
          "fontFamily": "Roboto",
          "fontSize": "16px",
          "fontWeight": "500",
          "lineHeight": "19px",
          "textAlign": "center",
          "textDecoration": "none",
          "textTransform": "none",
          "verticalAlign": "middle",
        }
      }
    >
       Misao Matimbo
    </span>
  </div>
</div>
`;

exports[`EventCounterParty withdraw matches snapshot 1`] = `
<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "WebkitBoxFlex": 1,
      "WebkitFlexBasis": "0%",
      "WebkitFlexGrow": 1,
      "WebkitFlexShrink": 1,
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
      "msFlexNegative": 1,
      "msFlexPositive": 1,
      "msFlexPreferredSize": "0%",
    }
  }
>
  <div
    className="css-text-901oao css-textOneLine-bfa6kz"
    dir="auto"
    style={
      Object {
        "color": "rgba(66,69,74,1.00)",
        "direction": "ltr",
        "fontFamily": "Roboto",
        "fontSize": "16px",
        "fontWeight": "normal",
        "lineHeight": "22px",
        "textAlign": "left",
        "textDecoration": "none",
        "textTransform": "capitalize",
      }
    }
  >
    <span
      className="css-text-901oao css-textHasAncestor-16my406"
      dir="auto"
      style={
        Object {
          "color": "rgba(66,69,74,1.00)",
          "direction": "ltr",
          "fontFamily": "Roboto",
          "fontSize": "10px",
          "fontWeight": "normal",
          "lineHeight": "14px",
          "marginRight": "3px",
          "textAlign": "center",
          "textDecoration": "none",
          "textTransform": "none",
          "verticalAlign": "middle",
        }
      }
    >
      From:
    </span>
    <span
      className="css-text-901oao css-textHasAncestor-16my406"
      dir="auto"
      style={
        Object {
          "color": "rgba(66,69,74,1.00)",
          "direction": "ltr",
          "fontFamily": "Roboto",
          "fontSize": "16px",
          "fontWeight": "500",
          "lineHeight": "19px",
          "textAlign": "center",
          "textDecoration": "none",
          "textTransform": "none",
          "verticalAlign": "middle",
        }
      }
    >
       Misao Matimbo
    </span>
  </div>
</div>
`;
